Sources/Security/Signing.swift

```diff
@@ -238,6 +246,15 @@ extension Signing.SignatureParameters {
 
 }
 
+private extension HTTPRequest.Path {
+
+    // Signatures remove percent encoding from the paths.
+    var unescapedPath: String {
+        return self.relativePath.removingPercentEncoding ?? ""
+    }
```

Comment:
This is slightly scary because of 
![image](https://github.com/RevenueCat/purchases-ios/assets/685609/d7a9be0f-54a3-4cc1-a286-7ce12b7e823e)

I'd vote to have the backend not unescape. Thoughts @bisho?

Reply:
Yeah, that would be my preference as well.

Reply:
Actually this is an implementation detail of the backend, I can handle it here without having to call `removingPercentEncoding`.